src/main/java/com/openshift/internal/restclient/capability/resources/PodExec.java

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.internal.restclient.capability.resources;
+
+import com.openshift.internal.restclient.DefaultClient;
+import com.openshift.internal.restclient.URLBuilder;
+import com.openshift.internal.restclient.okhttp.ResponseCodeInterceptor;
+import com.openshift.internal.restclient.okhttp.WebSocketAdapter;
+import com.openshift.restclient.IApiTypeMapper;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.capability.IStoppable;
+import com.openshift.restclient.capability.resources.IPodExec;
+import com.openshift.restclient.http.IHttpConstants;
+import com.openshift.restclient.model.IPod;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import okhttp3.ws.WebSocket;
+import okhttp3.ws.WebSocketCall;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class PodExec implements IPodExec {
```

Comment:
Consider extending from https://github.com/openshift/openshift-restclient-java/blob/master/src/main/java/com/openshift/internal/restclient/capability/AbstractCapability.java which will ensure the apigroups are checked for supportability